tests/i18n/test_compilation.py

```diff
@@ -154,17 +155,23 @@ def test_multiple_locales_excluded_with_locale(self):
 
 
 class CompilationErrorHandling(MessageCompilationTests):
-
-    LOCALE = 'ja'
-    MO_FILE = 'locale/%s/LC_MESSAGES/django.mo' % LOCALE
-
-    def setUp(self):
-        super(CompilationErrorHandling, self).setUp()
-        self.addCleanup(self.rmfile, os.path.join(self.test_dir, self.MO_FILE))
-
     def test_error_reported_by_msgfmt(self):
+        # po file contains wrong po formatting
+        mo_file = 'locale/ja/LC_MESSAGES/django.mo'
+        self.addCleanup(self.rmfile, os.path.join(self.test_dir, mo_file))
         with self.assertRaises(CommandError):
-            call_command('compilemessages', locale=[self.LOCALE], stdout=StringIO())
+            call_command('compilemessages', locale=['ja'], stdout=StringIO())
+
+        # po file contains invalid msgstr content
+        mo_file = 'locale/ko/LC_MESSAGES/django.mo'
+        self.addCleanup(self.rmfile, os.path.join(self.test_dir, mo_file))
+        # Various assertRaises on PY2 don't support unicode error messages
```

Comment:
Is it worth using `if six.PY2` and adding the Python 3 version with `assertRaisesMessage` so it's easy to cleanup when dropping PY2?